### ISSUE TITLE
All attached scripts break on the first HG jump to a region

### ISSUE THREAD

**Issue by lickx** [lickx/opensim#33]
All attached scripts stop working on teleport to hypergrid
All attached scripts lost state and will reset on return from hypergrid
Breaks content and makes the hypergrid totally unsuitable for production use IMHO.

script engine: xengine (yengine to be comfirmed)
mantis: http://opensimulator.org/mantis/view.php?id=8366
mantis: http://opensimulator.org/mantis/view.php?id=8049

Of note is that ArribaSim has this (according to the author partially) fixed

**Comment by lickx** [comment]
Also happens with yengine -> hg tp -> yengine

**Comment by lickx** [comment]
Follow up: abb1b1cce48023663bd8d85565c499c7a5fc5310

**Comment by lickx** [comment]
commit 4a043624833afe012b9174851f25513bfe2f335b solved this issue. If there will be an improved solution by someone else of myself then it will be replaced by that.

The cause? CompleteMovement() normally starts scripts (after making the agent a root agent) in all attachments. The edge case here is that if it is a HG visitor, no attachments have been transferred yet or are just starting to transfer, and this happens quite slowly no matter how fast the departure grid is; before the first attachment arrives, CompleteMovement() is already finished!

Turns out, in HGEntityTransferModule.cs incoming attachments are queued up (in a JobEngine) and call base.HandleIncomingSceneObject() in EntityTransferModule.cs. The latter function never started scripts in the attachments, which it now does. It would be bad to have CompleteMovement() wait for all attachments to have trickle in.

The relevant files:
Region/CoreModules/Framework/EntityTransfer/EntityTransferModule.cs
Region/CoreModules/Framework/EntityTransfer/HGEntityTransferModule.cs
Region/Framework/Scenes/ScenePresence.cs

**Comment by lickx** [comment]
Reopening this again as I encountered the issue again yesterday.

This is the test procedure which needs to be strictly followed or else you get false results:

A hypergrid alt needs to teleport to us with attachments which hold scripts which have not been previously seen/cached in our sim. Best to create a new short test script in a cube every time and attach it, then TP.

The outcome should be that when we have teleported, the script is running and events (touched, changed etc) working.

I'm working on finding a better solution to this issue.
